agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -0,0 +1,122 @@
+package grpcclient
+
+import (
+	"context"
+	"os"
+	"time"
+
+	pb "github.com/aws/amazon-ecs-agent/agent/taskresource/grpcclient/credentialsfetcher"
+	"github.com/cihub/seelog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type CredentialsFetcherClient struct {
+	conn    *grpc.ClientConn
+	timeout time.Duration
+}
+
+// GetGrpcClientConnection() returns grpc client connection
+func GetGrpcClientConnection() (*grpc.ClientConn, error) {
+	address, err := GetSocketAddress()
+	if err != nil {
+		seelog.Errorf("could not find path to credentials fetcher host dir : %v", err)
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		seelog.Errorf("could not initialize client connection %v", err)
+		return nil, err
+	}
+	return conn, nil
+
+}
+
+// GetSocketAddress() returns the credentials-fetcher socket dir
+func GetSocketAddress() (string, error) {
+	credentialsfetcherHostDir := os.Getenv("CREDENTIALS_FETCHER_HOST_DIR")
```

Comment:
Can we add this new environment variable to README with some details if value of this environment can be customized?
> FYI. We later will also need to add this config variable to [Amazon ECS container agent configuration](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-config.html#ecs-agent-availparam).

Reply:
yes I will in the feature branch pull request, I already have this in the other PR